libs/BLEKit/source/CoreGap.cpp

```diff
@@ -7,12 +7,12 @@ using namespace ble;
 
 void CoreGap::setDefaultAdvertising()
 {
-	_advertising_data_builder.setAppearance(adv_data_appearance_t::GENERIC_HEART_RATE_SENSOR);
-	_advertising_data_builder.setFlags();
-	_advertising_data_builder.setManufacturerSpecificData({{0x2A, 0x2B, 0x2C, 0x2D}});
-	_advertising_data_builder.setAdvertisingInterval(adv_interval_t::min());
-	_advertising_data_builder.setServiceData(GattService::UUID_BATTERY_SERVICE, {{0x42}});
-	_advertising_data_builder.setServiceData(leka::service::commands::uuid, {{0}});
+	auto default_advertising_data = AdvertisingData {};
+
+	_advertising_data_builder.setName(default_advertising_data.name);
+	_advertising_data_builder.setServiceData(
+		leka::service::commands::uuid,	 // TODO: commands::uuid only for compatibility with LekaApp
+		{default_advertising_data.getData().data(), default_advertising_data.getDataSize()});
```

Comment:
with my new implementation, no need for `getData().data()`

```suggestion
		{default_advertising_data.data(), default_advertising_data.size()});
```

Reply:
It compiles but clangd shows this error

```
Non-constant-expression cannot be narrowed from type 'std::array::size_type' (aka 'unsigned long') to 'mbed::Span<type-parameter-0-0, -1>::index_type' (aka 'long') in initializer list (fix available)clang(-Wc++11-narrowing)
```